app/models/ability.rb

```diff
@@ -146,57 +148,76 @@ def signed_in_with_roles(user)
     end
   end
 
-  def signed_in_with_organizer_role(user)
-    # ids of all the conferences for which the user has the 'organizer' role
-    conf_ids_for_organizer = Conference.with_role(:organizer, user).pluck(:id)
-
-    can :manage, Resource, conference_id: conf_ids_for_organizer
-    can [:new, :create], Conference if user.has_role?(:organizer, :any)
-    can :manage, Conference, id: conf_ids_for_organizer
-    can :manage, Splashpage, conference_id: conf_ids_for_organizer
-    can :manage, Contact, conference_id: conf_ids_for_organizer
-    can :manage, EmailSettings, conference_id: conf_ids_for_organizer
-    can :manage, Campaign, conference_id: conf_ids_for_organizer
-    can :manage, Target, conference_id: conf_ids_for_organizer
+  def signed_in_with_organization_admin_role(user)
+    org_ids_for_organization_admin = Organization.with_role(:organization_admin, user).pluck(:id)
+
+    can :manage, Organization, id: org_ids_for_organization_admin
+    can :new, Conference
+    can :manage, Conference, organization_id: org_ids_for_organization_admin
+    conf_ids_for_organization_admin = []
+    org_ids_for_organization_admin.each do |org_id|
+      conf_ids_for_organization_admin += Organization.find(org_id).conferences.pluck(:id)
+    end
+    can [:index, :show], Role
+    can [:edit, :update], Role do |role|
+      role.resource_type == 'Organization' && (org_ids_for_organization_admin.include? role.resource_id)
+    end
+    signed_in_with_organizer_role(user, conf_ids_for_organization_admin)
+  end
+
+  def signed_in_with_organizer_role(user, conf_ids_for_organization_admin = [])
```

Comment:
I would just rename `conf_ids_for_organization_admin_and_organizer` as `conf_ids`... such a long name makes the code hard to read, and it's obvious anyway.